sys/log/src/log.c

```diff
@@ -103,20 +103,27 @@ log_append(struct log *log, uint16_t module, uint16_t level, void *data,
     int rc;
     struct os_timeval tv;
 
-    log->l_index++;
-
     ue = (struct log_entry_hdr *) data;
 
+    /* Resetting index every millisecond */
+    if (g_log_info.li_timestamp > ue->ue_ts + 1000) {
+        g_log_info.li_index = 0;
+    }
+
+    g_log_info.li_index++;
+
```

Comment:
The index is now now treated as a secondary key instead of a primary key for the logs. Each log doesn't have it's own index anymore. It is a global index. Spoke to Will, Chris and Paul. The reason for making it a secondary key is that the index would roll over only after 1 millisecond elapses since the last timestamp. If you see multiple logs from different modules, you can actually relate the events looking at the index for the same timestamp. Also, the belief is we would not ever see more than 255 events being logged in a millisecond. Hope this makes sense.